src/Form.php

```diff
@@ -0,0 +1,378 @@
+     */
+    public function getAttributeLabel(string $attribute): string
+    {
+        $this->attributesLabels = $this->getAttributesLabels();
+
+        return isset($this->attributesLabels[$attribute])
+            ? $this->attributesLabels[$attribute]
+            : $this->generateAttributeLabel($attribute);
+    }
+
+    /**
+     * Returns the form name that this model class should use.
+     *
+     * The form name is mainly used by {\Yiisoft\Form\FormWidget} to determine how to name the input fields for the
+     * attributes in a model. If the form name is "A" and an attribute name is "b", then the corresponding input name
+     * would be "A[b]". If the form name is an empty string, then the input name would be "b".
+     *
+     * The purpose of the above naming schema is that for forms which contain multiple different models, the attributes
+     * of each model are grouped in sub-arrays of the POST-data and it is easier to differentiate between them.
+     *
+     * By default, this method returns the model class name (without the namespace part) as the form name. You may
+     * override it when the model is used in different forms.
+     *
+     * @return string|null the form name of this model class.
+     *
+     * {@see load()}
+     */
+    public function getFormname(): ?string
+    {
+        return null;
+    }
+
+    /**
+     * Returns the errors for all attributes or a single attribute.
+     *
+     * @param string|null $attribute attribute name. Use null to retrieve errors for all attributes.
+     * @property array An array of errors for all attributes. Empty array is returned if no error. The result is a
+     * two-dimensional array. {@see getErrors()} for detailed description.
+     * @return array errors for all attributes or the specified attribute. Empty array is returned if no error.
+     *
+     * Note that when returning errors for all attributes, the result is a two-dimensional array, like the following:
+     *
+     * ```php
+     * [
+     *     'username' => [
+     *         'Username is required.',
+     *         'Username must contain only word characters.',
+     *     ],
+     *     'email' => [
+     *         'Email address is invalid.',
+     *     ]
+     * ]
+     * ```
+     *
+     * {@see getFirstErrors()}
+     * {@see getFirstError()}
+     */
+    public function getErrors(?string $attribute = null): array
+    {
+        if ($attribute === null) {
+            return $this->errors === null ? [] : $this->errors;
+        }
+
+        return isset($this->errors[$attribute]) ? $this->errors[$attribute] : [];
+    }
+
+    /**
+     * Returns the first error of the specified attribute.
+     *
+     * @param string $attribute attribute name.
+     *
+     * @return string|null the error message. Null is returned if no error.
+     *
+     * {@see getErrors()}
+     * {@see getFirstErrors()}
+     */
+    public function getFirstError(string $attribute): string
```

Comment:
Maybe we need to return `null` when errors are absent?

Reply:
done.